ext/nnstreamer/tensor_decoder/tensordec-tensor_region.c

```diff
@@ -0,0 +1,758 @@
+    if (a->valid == TRUE) {
+      for (j = i + 1; j < boxes_size; j++) {
+        detected_object *b = &g_array_index (results, detected_object, j);
+        if (b->valid == TRUE) {
+          if (iou (a, b) > threshold) {
+            b->valid = FALSE;
+          }
+        }
+      }
+    }
+  }
+
+  i = 0;
+  do {
+    detected_object *a = &g_array_index (results, detected_object, i);
+    if (a->valid == FALSE)
+      g_array_remove_index (results, i);
+    else
+      i++;
+  } while (i < results->len);
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static GstFlowReturn
+tr_decode (void **pdata, const GstTensorsConfig *config,
+    const GstTensorMemory *input, GstBuffer *outbuf)
+{
+  tensor_region *trData = *pdata;
+  GstMapInfo out_info;
+  GstMemory *out_mem;
+  GArray *results = NULL;
+  const guint num_tensors = config->info.num_tensors;
+  gboolean need_output_alloc = gst_buffer_get_size (outbuf) == 0;
+  const size_t size = (size_t) 4 * trData->num; /**4 field per block */
+
+  g_assert (outbuf);
+  /**Ensure we have outbuf properly allocated */
+  if (need_output_alloc) {
+    out_mem = gst_allocator_alloc (NULL, size, NULL);
+  } else {
+    if (gst_buffer_get_size (outbuf) < size) {
+      gst_buffer_set_size (outbuf, size);
+    }
+    out_mem = gst_buffer_get_all_memory (outbuf);
+  }
+  if (!gst_memory_map (out_mem, &out_info, GST_MAP_WRITE)) {
+    ml_loge ("Cannot map output memory / tensordec-tensor_region.\n");
+    goto error_free;
+  }
+
+  /** reset the buffer with 0 */
+  memset (out_info.data, 0, size);
+  if(trData->mode == MOBILENET_SSD_BOUNDING_BOX){
+    const GstTensorMemory *boxes, *detections = NULL;
+    properties_MOBILENET_SSD *data = &trData->mobilenet_ssd;
+
+    g_assert (num_tensors >= MOBILENET_SSD_MAX_TENSORS);
+    results = g_array_sized_new (FALSE, TRUE, sizeof (detected_object), 100);
+
+    boxes = &input[0];
+    if (num_tensors >= MOBILENET_SSD_MAX_TENSORS) /**lgtm[cpp/constant-comparison] */
+      detections = &input[1];
+    
+    switch (config->info.info[0].type) {
+      _get_objects_mobilenet_ssd_ (uint8_t, _NNS_UINT8);
+      _get_objects_mobilenet_ssd_ (int8_t, _NNS_INT8);
+      _get_objects_mobilenet_ssd_ (uint16_t, _NNS_UINT16);
+      _get_objects_mobilenet_ssd_ (int16_t, _NNS_INT16);
+      _get_objects_mobilenet_ssd_ (uint32_t, _NNS_UINT32);
+      _get_objects_mobilenet_ssd_ (int32_t, _NNS_INT32);
+      _get_objects_mobilenet_ssd_ (uint64_t, _NNS_UINT64);
+      _get_objects_mobilenet_ssd_ (int64_t, _NNS_INT64);
+      _get_objects_mobilenet_ssd_ (float, _NNS_FLOAT32);
+      _get_objects_mobilenet_ssd_ (double, _NNS_FLOAT64);
+      default:
+        g_assert (0);
+    }
+
+    nms (results, data->params[MOBILENET_SSD_PARAMS_IOU_THRESHOLD_IDX]);
+  } else {
+    GST_ERROR ("Failed to get output buffer, unknown mode %d.", trData->mode);
+    goto error_unmap;
+  }
+  finalize (&out_info, trData, results);

```

Comment:
The info pad of `tensor_crop` has capability for flexible tensor stream (other/tensors-flexible).
You should transform crop info to flexible tensors here.